### PR TITLE
Use Jasmine for testing

### PR DESCRIPTION
**Pull request by ryzokuken**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,9 +1,8 @@
 {
     "env": {
-        "browser": true,
-        "commonjs": true,
```
Comment (ananyo2012) [inline .eslintrc]: We are using commonjs conventions for module exports/imports right?

Comment (ryzokuken) [inline .eslintrc]: I think those are included in the nodejs environment.

http://eslint.org/docs/user-guide/configuring#specifying-environments says that commonjs - CommonJS global variables and CommonJS scoping (use this for browser-only code that uses Browserify/WebPack).``

**Comment by ryzokuken** [comment]
@ananyo2012 That's too much configuration. I instead used a package that does that for us. Its called `jasmine-node` (https://github.com/mhevery/jasmine-node)

File: .eslintrc (hunk 1 above)
Comment (ananyo2012) [inline .eslintrc]: Ok seems compliant with documentation.

**Comment by ananyo2012** [comment]
@ryzokuken I strongly recommend you to use the latest jasmine npm package as that's the one supported by jasmine project itself. Check https://github.com/jasmine/jasmine#installation. The `jasmine-node` package was written way before and is also isn't updated for a long time. And there isn't much setup involved, just a `spec/support/jasmine.json` file which can be automatically generated using `jasmine init` command. Also running tests is simple with just the `jasmine` command.

**Comment by ryzokuken** [comment]
@ananyo2012 Okay, I will make another branch for this. I wanted to discuss these things, which is why I added the `DO NOT MERGE` label on this one. Meanwhile, could you take a look at the other PR?

**Comment by ananyo2012** [comment]
Is this a stale branch now? Should I close this one?

**Comment by ryzokuken** [comment]
Yes, this needs to be closed now.

**Pull request closed without merging**